File: frontend/src/types/drug.ts
import type { StockStatus } from "./enums";

export interface DrugListItem {
  id: number;
  code: string;
  name: string;
  strength: string | null;
  dosage_form: string | null;
  is_active: boolean;
  quantity_on_hand: number;
  stock_status: StockStatus;
}

export interface DrugDetail extends DrugListItem {
  low_stock_threshold: number;
  reorder_threshold: number;
  last_restocked_at: string | null;
}

File: backend/app/schemas/drug.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.domain.enums import StockStatus


class DrugListItem(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    code: str
    name: str
    strength: str | None
    dosage_form: str | None
    is_active: bool
    quantity_on_hand: int
    stock_status: StockStatus


class DrugDetail(DrugListItem):
    low_stock_threshold: int
    reorder_threshold: int
    last_restocked_at: datetime | None
